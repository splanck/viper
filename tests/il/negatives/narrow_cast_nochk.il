il 0.1.2

func @narrow_cast(%f:f64) -> i64 {
entry(%f:f64):
  %t0 = fptosi %f
  ret %t0
}
