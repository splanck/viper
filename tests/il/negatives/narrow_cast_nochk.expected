error: narrow_cast:entry: %2 = fptosi %t1: fp to integer narrowing must use cast.fp_to_si.rte.chk (rounds to nearest-even and traps on overflow)
